### DISCUSSION TITLE
Remember this browser option has a lot of steps

### DISCUSSION THREAD

**Discussion by neilmcd** [neos-sdi/adfsmfa#249] [Ideas]
Would it be possible to rethink where in the login process the 'Stop being asked for 7 days' option is located? 

At the moment users need to select 'Access my options after authentication', click sign in, then click another sign in button, then select if they want to stop being asked for MFA for 7 days. It feels like a lot of steps.

It would be good if there was a simple option to remember the browser when typing in the code to reduce the number of clicks needed (see image for example). Then the cookie gets set at the first login button click.

![2022-09-22_22-47-32](https://user-images.githubusercontent.com/81166/191858358-2f84b1dc-42d6-430e-ac69-cb64f52054bc.png)

**Comment by redhook62** [comment]
Hi @neilmcd 

I understand your request, however the current functionality is inspired by the Microsoft Login on AAD.

Positioning this request before entering the MFA is not desirable for security, moreover the screen already contains a lot of information. some authentication modes (Two Ways) do not go through an intermediate screen.

On the other hand, there is currently no obligation to go through "manage my options", 
if you don't have this working, please detail your configuration, so that we can do some tests.

Regards

**Comment by neilmcd** [comment]
Hi, thanks for the response.

> On the other hand, there is currently no obligation to go through "manage my options",

As far as I can see, you must go through 'Manage my options' if you want to see the 'Stop being asked for 7 days' option. Do you mean that it should appear without doing that and something in my config is stopping it?

**Comment by redhook62** [comment]
Hi, 

The suspension options are configured by the administrators (via the MMC console), then this is offered to all users at the end of authentication process.
If this does not work for you, please send us your configuration file as indicated: Export-MFASystemConfiguration. 
you can anonymize whatever you want and send it to: adfsmfa@gmail.com.
maybe there's a bug ?

**Comment by redhook62** [comment]
Hi @neilmcd 

Well, after a battery of tests, and only on ADFS 2016 (therefore not reproduced on 2012r2, 2019 and 2022 in paginated mode or not), we have managed to reproduce your problem.

So, it seems to be a bug. we'll look for a fix next week.

If you wish, for the community, you can post an issue describing your problem.

regards

redhook

**Comment by neilmcd** [comment]
That's great, thanks for looking into it so quickly! I'll raise it as an issue.
